src/components/MenuBar/ToggleGridViewButton/ToggleGridViewButton.tsx

```diff
@@ -1,16 +1,28 @@
 import React from 'react';
 import GridOnIcon from '@material-ui/icons/GridOn';
-import GridOffIcon from '@material-ui/icons/GridOff';
 import IconButton from '@material-ui/core/IconButton';
-
+import { ViewModeArray, Settings } from '../../../state/settings/settingsReducer';
 import { useAppState } from '../../../state';
 
 export default function ToggleGridViewButton() {
-    const { gridView, setGridView } = useAppState();
+  const { settings, dispatchSetting } = useAppState();
+  const handleChange = () => {
+    const indexOfCurrMode = ViewModeArray.indexOf(settings.viewMode);
+    const indexOfNextMode = indexOfCurrMode + 1 < ViewModeArray.length ? indexOfCurrMode + 1 : 0;
+    const nextView = ViewModeArray[indexOfNextMode];
+    dispatchSetting({
+      name: 'viewMode' as keyof Settings,
+      value: nextView,
+    });
+  };
 
-    return (
-        <IconButton onClick={() => setGridView(!gridView)}>
-            {gridView ? <GridOffIcon /> : <GridOnIcon />}
-        </IconButton>
-    );
+  return (
+    <IconButton
+      onClick={() => {
```

Comment:
change the onClick={...} to:
onClick={handleChange}

Reply:
done